compiler/rustc_resolve/src/late/diagnostics.rs

```diff
@@ -250,17 +250,34 @@ impl<'a: 'ast, 'ast> LateResolutionVisitor<'a, '_, 'ast> {
                 .map_or_else(String::new, |res| format!("{} ", res.descr()));
                 (mod_prefix, format!("`{}`", Segment::names_to_string(mod_path)), None)
             };
+
+            let (fallback_label, override_sugg) = if path_str == "async"
+                && expected.starts_with("struct")
+            {
+                ("`async` blocks are only allowed in Rust 2018 or later".to_string(), suggestion)
+            } else {
+                // check if we are in situation of typo like `True` instead of `true`.
+                let override_siggestion =
+                    if ["true", "false"].contains(&item_str.to_string().to_lowercase().as_str()) {
```

Comment:
I think this would be true if `item_str` is like `tRue` or `trUE`. Could you change this to something like:

```suggestion
                    let sugg = match item_str.as_str() {
                        "True" => Some(kw::True),
                        "False" => Some(kw::False),
                        _ => None,
                    };
                    if let Some(sugg) = sugg {
```

Reply:
in this case, we do catch cases like "TRue" and "TruE"

Reply:
However, I can keep `.to_string().to_lowercase().as_str()` and us the match with the case in lower case.

I like your refactoring

Reply:
Sorry. What I really mean is that this condition should not be true if `item_str` is like `tRue`, `trUE`, `TRue` or `TruE`.

Reply:
A wait! my screen reader jump a line, with this

> I think this would be true if item_str is like tRue or trUE. Could you change this to something like:

Do you think that it is not good normalize all the typo possibility? but just include `True` and `False`?

Reply:
Yes, I do. I think rustc should suggest `true` or `false` when using `True` and `False` instead of `true` and `false` because they are used for other languages such as Python.

Reply:
I think it's okay that this suggests using `true` for `tRue`/`truE`/`tRUE`/etc and not just `True` - the user might not have been trying to use another language's syntax, but it would still help fix what is likely to be a typo.

Reply:
I vote for keeping the typo fixing, this could also use in code generation, where the compiler generates rust and make a stupid typo into it.